src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -12,6 +12,12 @@ import scala.collection.JavaConverters._
 import scala.language.implicitConversions
 
 abstract class GRBase extends Serializable {
+  def variant: TVariant
+
+  def locus: TLocus
+
+  def interval: TInterval
```

Comment:
The definitions of these in the derived classes are the same, so I think you can just make these `val`s and define them here.